openslides_backend/action/mixins/delegation_based_restriction_mixin.py

```diff
@@ -51,43 +51,38 @@ def check_delegator_restriction(
         """
         if not len(meeting_ids):
             return []
-        data = self.datastore.get_many(
-            [
-                GetManyRequest(
-                    "user",
-                    [self.user_id],
-                    ["meeting_user_ids"],
-                ),
-                GetManyRequest(
-                    "meeting",
-                    meeting_ids,
-                    [restriction, "users_enable_vote_delegations"],
+
+        meeting_users = self.datastore.filter(
+            "meeting_user",
+            And(
+                Or(
+                    FilterOperator("meeting_id", "=", meeting_id)
+                    for meeting_id in meeting_ids
                 ),
-            ]
+                FilterOperator("user_id", "=", self.user_id),
+                FilterOperator("vote_delegated_to_id", "!=", None),
+            ),
+            ["meeting_id"],
+            lock_result=False,
         )
-        operator = data["user"][self.user_id]
-        if operator.get("meeting_user_ids"):
-            meeting_users = self.datastore.filter(
-                "meeting_user",
-                And(
-                    Or(
-                        FilterOperator("meeting_id", "=", meeting_id)
-                        for meeting_id in meeting_ids
-                    ),
-                    Or(
-                        FilterOperator("id", "=", meeting_user_id)
-                        for meeting_user_id in operator["meeting_user_ids"]
+        if len(meeting_users):
+            delegation_meeting_ids = [
+                meeting_user["meeting_id"] for meeting_user in meeting_users.values()
+            ]
+            data = self.datastore.get_many(
```

Comment:
I suggest calling this variable `delegation_meetings` or `delegation_meetings_data`.

Reply:
Done